mkosi/__init__.py

```diff
@@ -1449,30 +1449,41 @@ def delete_whiteout_files(path: Path) -> None:
                 os.unlink(entry.path)
 
 
-def do_mount(
+@contextlib.contextmanager
+def mount(
         what: PathString,
         where: Path,
+        operation: Optional[str] = None,
         options: Sequence[str] = (),
         type: Optional[str] = None,
         read_only: bool = False,
-) -> None:
+) -> Iterator[Path]:
     os.makedirs(where, 0o755, True)
 
     if read_only:
         options = ["ro", *options]
 
-    cmd: List[PathString] = ["mount", "-n", what, where]
+    cmd: List[PathString] = ["mount", "-n"]
+
+    if operation:
+        cmd += [operation]
+
+    cmd += [what, where]
 
     if type:
         cmd += ["-t", type]
 
     if options:
         cmd += ["-o", ",".join(options)]
 
-    run(cmd)
+    try:
+        run(cmd)
+        yield where
+    finally:
+        run(["umount", "-n", where])
```

Comment:
```suggestion
        run(["umount", "--no-mtab", where])
```
since it's nicer to follow and easier to look up when one doesn't have the umount options in one's head.

The idea here is to not use `--recursive`, since when exiting the stack all the exiting umounts will be popped of the stack, right? A `--recursive` wouldn't hurt, though, and might clean up things even if an umount fails (since I think the `__exit__`s from the context manager will be called regardless even on an exception)

Reply:
I figure since it works without --recursive that's also nice though, since it lets us know when we're forgetting to unmount something.

Reply:
Added it back, probably safer to keep it there